src/relax/ir/block_builder.cc

```diff
@@ -128,9 +131,19 @@ class BlockBuilderImpl : public BlockBuilderNode {
       }
       GlobalVar gvar = GlobalVar(func_name);
 
-      ICHECK(func->checked_type_.defined())
-          << "The function to be added does not have checked_type_.";
-      gvar->checked_type_ = func->checked_type_;
+      StructInfo finfo;
+      if (func->struct_info_.defined()) {
+        finfo = GetStructInfo(func);
+      } else if (auto* prim_func = func.as<tir::PrimFuncNode>()) {
+        // NOTE: use a slightly different struct info than checked type
+        // in PrimFunc so handle can turn into Tensor.
+        // TODO(relax-team): add fine-grained PrimFunc struct info signature generation.
+        finfo = FuncStructInfo::OpaqueFunc(StructInfoFromType(prim_func->ret_type));
+      } else {
+        finfo = StructInfoFromType(func->checked_type_);
```

Comment:
Use `checked_type()` which has a check on defined or not.
```suggestion
        finfo = StructInfoFromType(func->checked_type());
```